src/presentation/controller/addressLookUp/Controller.ts

```diff
@@ -423,12 +429,17 @@ class AddressLookUpController extends AbstractController {
     response.redirect(pageRouting.nextUrl);
   }
 
+  // TODO As each pageType can have a cache key, add the key to each 'routing' object for each page and remove the if/elses
   private getCacheKey(pageType: any) {
     let cacheKey = "";
     if (this.isPrincipalPlaceOfBusinessPage(pageType)) {
       cacheKey = this.PRINCIPAL_PLACE_OF_BUSINESS_ADDRESS_CACHE_KEY;
     } else if (this.isRegisteredOfficeAddressPage(pageType)) {
       cacheKey = this.REGISTERED_OFFICE_ADDRESS_CACHE_KEY;
+    } else if (pageType === AddressLookUpPageType.territoryChoiceGeneralPartnerUsualResidentialAddress) {
```

Comment:
Was this setting of the cache key for URA missing then? Assume it was as it looks like it's being added here along with the Principal key.